ckanext/xloader/plugin.py

```diff
@@ -64,16 +68,29 @@ def configure(self, config_):
                     )
                 )
 
-    # IResourceUrlChange
+    # IDomainObjectModification
+
+    def notify(self, entity, operation):
+        # type: (ckan.model.Package|ckan.model.Resource, DomainObjectOperation) -> None
+        """
+        Runs before_commit to database for Packages and Resources.
+        We only want to check for changed Resources for this.
+        We want to check if values have changed, namely the url and the format.
+        See: ckan/model/modification.py.DomainObjectModificationExtension
+        """
+        if operation != DomainObjectOperation.changed or not isinstance(entity, Resource):
+            return
+
+        if _should_remove_unsupported_resource_from_datastore(entity):
```

Comment:
Wouldn't this be a better place to extract the `dict`?

Reply:
Ah I see what you mean. Haha don't know how I missed that. As you are already calling `resource_show` ti get the resource dict below.

I have moved this line and reworked the `_should_remove_unsupported_resource_from_datastore` a bit.

I initially had the entire resource dict going into the job callback, but I had issues with multiple updates to a resource in quick succession. So I ended up only passing the resource ID to the callback, and doing the extra `resource_show` in there to be sure that the job will have the actual current data for the resource

Reply:
Yeah, don't pass entire dictionaries to jobs if you can possibly avoid it. With large job numbers that swamps the memory of the Redis cache.